pkg/util/diff/diff.go

```diff
@@ -101,3 +113,142 @@ func LoadFile(input, location string) (ytbx.InputFile, error) {
 		Documents: documents,
 	}, nil
 }
+
+// maskSensitiveData masks the sensitive data with placeholders before generating
+// the diff report.
+func maskSensitiveData(oldData, newData interface{}) (interface{}, interface{}) {
+	from, ok1 := oldData.(*v1.Resource)
+	to, ok2 := newData.(*v1.Resource)
+
+	// Return directly if oldData or newData can not be transferred into v1.Resource.
+	if !(ok1 && ok2) {
+		return oldData, newData
+	}
+
+	// Record whether we need to mask the old or the new object and the 'data'
+	// and 'stringData' attributes of the secret resource.
+	maskOld, maskNew := false, false
+	fromSecData, toSecData := map[string]interface{}{}, map[string]interface{}{}
+	fromSecStrData, toSecStrData := map[string]interface{}{}, map[string]interface{}{}
+
+	// Prepare the masked old secret resource and masked new secret resource.
+	maskedOldData, maskedNewData := &v1.Resource{}, &v1.Resource{}
+
+	// Check if the resource type is Kubernetes Secret.
+	if from != nil {
+		fromKind, ok := from.Attributes["kind"].(string)
```

Comment:
What if `from.Attributes["kind"]` is nil? Converting nil to string will cause a panic